lib/puppet/util/windows/sid.rb

```diff
@@ -79,7 +84,8 @@ def name_to_principal(name)
 
       raw_sid_bytes ? Principal.lookup_account_sid(raw_sid_bytes) : Principal.lookup_account_name(name)
     rescue
-      nil
+      #if we received a SID and we were unable to resolve it, pass it back as unresolved SID
+      raw_sid_bytes ? unresolved_principal(name, raw_sid_bytes) : nil
```

Comment:
Related to this check, we tried to fix a similar behavior in groupadd provider https://github.com/puppetlabs/puppet/pull/8013, there the conclusions were:
 - member_valid? should check validity from syntax/formatting point of view only, due to
   - if member_valid? raises, it will be treated like a manifest syntax error and will not be applied at all
   - you will not be able to create a user and add it to a group in the same manifest
- at least for `groupadd` provider, the `member_valid?` is expected to throw if validation fail, it's return value is ignored. If it is the same here, the validation will always pass as `name_to_principal` rescues everything and returns `nil`: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/sid.rb#L81